utils/globalvar_test.go

```diff
@@ -0,0 +1,62 @@
+package utils
+
+import (
+	"os"
+	"testing"
+)
+
+func TestParseIntEnvVar(t *testing.T) {
+	tests := []struct {
+		name         string
+		varName      string
+		varValue     string
+		defaultValue int
+		expected     int
+		expectError  bool
+	}{
+		{
+			name:         "environment variable not set",
+			varName:      "TEST_ENV_VAR_NOT_SET",
+			defaultValue: 10,
+			expected:     10,
+			expectError:  false,
+		},
+		{
+			name:         "environment variable set with valid integer",
+			varName:      "TEST_ENV_VAR_VALID",
+			varValue:     "20",
+			defaultValue: 10,
+			expected:     20,
+			expectError:  false,
+		},
+		{
+			name:         "environment variable set with invalid integer",
+			varName:      "TEST_ENV_VAR_INVALID",
+			varValue:     "invalid",
+			defaultValue: 10,
+			expected:     10,
+			expectError:  true,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			if tt.varValue != "" {
+				os.Setenv(tt.varName, tt.varValue)
+			} else {
+				os.Unsetenv(tt.varName)
+			}
```

Comment:
issue: an environment variable leaks after running this test


Since there is no cleanup of variables after setting them up, an environment variable will leak into the runner system. Which is why it’s a good idea to use test doubles for environment variables, not actual the actual live ones. If we do want the live variables, just use the function provided for testing:
```suggestion
			if tt.varValue != "" {
				t.Setenv(tt.varName, tt.varValue)
			} else {
				t.Setenv(tt.varName, "")
			}
```

Reply:
I implemented it with defer